### DISCUSSION TITLE
pixel editor - eraser tool over black areas instead of white

### DISCUSSION THREAD

**Discussion by northernmix** [sn4k3/UVtools#164] [Ideas]
Hello, been using your program for a couple of weeks and love the pixel editor!

But I was wondering if it's possible to add a function to the "Eraser" tool: that allows me to shift+click over a BLACK pixel area (that is surrounded by white pixels) yet still allow me to set a value in "layers depth below" to delete pixels below with the same fill shape. Essentially its what the Eraser tool already does but allows me to select black pixels instead of white pixels.

The reason for this idea is that sometimes my models have complex intersecting polygons causing a layer inside a hollowed mesh to flash on then disappear in the next layer after being sliced. Currently, I'm working around this by going to the "Drawing" tool, slowly tracing the black space in the layer above by hand with a "shift+right click" while setting "layers depth below" to 1 to remove the pixels.

Thanks!

**Comment by sn4k3** [comment]
I think i understand what you need, it's somekind of the tool solidify on ROI / Resin trap fixer on ROI? If you have some image samples you can post please do so i can think better on that.

**Reply by northernmix** [reply]
Ok I've attached some images showing what I am doing :) The resin trap fixer doesn't quite work for my case because the unwanted pixel flash can occur even if the model has drainage holes above and/or below the pixels I need to remove.

Explanation:
Image 1 = shows layer 649 looking ok
Image 2 = shows layer 650 which suddenly becomes solid and poses a problem
Image 3 = shows layer 651 looking ok again
Image 4 = shows what I do to fix the problem, by tracing the black area with a brush tool and setting "layers depth below" to 1 to fix 650 below

![1-layer649](https://user-images.githubusercontent.com/80502699/110896602-0d60c480-8350-11eb-92b7-16aa3ab644e5.jpg)
![2-layer650](https://user-images.githubusercontent.com/80502699/110896620-1356a580-8350-11eb-86a9-8a0edbeff307.jpg)
![3-layer651](https://user-images.githubusercontent.com/80502699/110896634-16ea2c80-8350-11eb-9ee9-0ff6455c9170.jpg)
![4-layer651_drawing](https://user-images.githubusercontent.com/80502699/110896646-1baee080-8350-11eb-904a-4b39b0644cd4.jpg)

**Reply by sn4k3** [reply]
Ok i see, you want to remove that middle white blob to open the path right? If so the only problem with eraser is all details on the middle will be wiped as it fill contours, so that two dots (supports?) would be erased too

**Reply by sn4k3** [reply]
Try the following manupulation:

1) Go to layer 649
2) Select that black rectangle portion with ROI (SHIFT + click and drag)
2) Go Arithmetic tool
3) 650=649
4) Hit apply and test

ROI like:
![image](https://user-images.githubusercontent.com/113281/110898407-183b3b00-82f7-11eb-9c39-59745f564f29.png)

**Reply by northernmix** [reply]
Yes, I tried it and it does what I needed to do, thankyou very much. This automation has saved me a lot of time.
Thankyou :) :)

**Reply by sn4k3** [reply]
Glad to help. I will also try to implement a eraser on black

**Reply by sn4k3** [reply]
v2.7.0 can now make use of Masks to ease this task

**Comment by northernmix** [comment]
Wow thank you very much, this function will greatly help my work flow, I
will update to the new version :)

Much appreciated,
Sidney
